docs/Serial.md

```diff
@@ -56,86 +56,151 @@ e.g. after configuring a port for GPS enable the GPS feature.
+| 2. Serial Port Function        |         | serialPortFunction_e     |
+| 3. Configuration/MSP baud rate | 115200  | baudrate_e               |
+| 4. Sensor Input baud rate      | 57600   | gpsbaudRate_e            |
+| 5. Telemetry baud rate         | AUTO    | baudRate_e               |
+| 6. Peripherals baud rate       | 115200  | baudrate_e               |
+
+### 1. Serial Port Identifier
+
+| Identifier                 | Value |
+| -------------------------- | ----- |
+| SERIAL_PORT_NONE           | -1    |
+| SERIAL_PORT_USART1         | 0     |
+| SERIAL_PORT_USART2         | 1     |
+| SERIAL_PORT_USART3         | 2     |
+| SERIAL_PORT_UART4          | 3     |
+| SERIAL_PORT_UART5          | 4     |
+| SERIAL_PORT_USART6         | 5     |
+| SERIAL_PORT_USART7         | 6     |
+| SERIAL_PORT_USART8         | 7     |
+| SERIAL_PORT_LPUART1        | 8     |
+| SERIAL_PORT_USB_VCP        | 20    |
+| SERIAL_PORT_SOFTSERIAL1    | 30    |
+| SERIAL_PORT_SOFTSERIAL2    | 31    |
+| SERIAL_PORT_IDENTIFIER_MAX | 31    |
+
+Note: `SERIAL_PORT_IDENTIFIER_MAX` = `SERIAL_PORT_SOFTSERIAL2`
+
+### 2. Serial Port Function
+
+ | Function                     | Value |
+ | ---------------------------- | ----- |
+ | FUNCTION_NONE                | 0     |
+ | FUNCTION_MSP                 | 1     |
+ | FUNCTION_GPS                 | 2     |
+ | FUNCTION_TELEMETRY_FRSKY_HUB | 4     |
+ | FUNCTION_TELEMETRY_HOTT      | 8     |
+ | FUNCTION_TELEMETRY_LTM       | 16    |
+ | FUNCTION_TELEMETRY_SMARTPORT | 32    |
+ | FUNCTION_RX_SERIAL           | 64    |
+ | FUNCTION_BLACKBOX            | 128   |
+ | FUNCTION_TELEMETRY_MAVLINK   | 512   |
+ | FUNCTION_ESC_SENSOR          | 1024  |
+ | FUNCTION_VTX_SMARTAUDIO      | 2048  |
+ | FUNCTION_TELEMETRY_IBUS      | 4096  |
+ | FUNCTION_VTX_TRAMP           | 8192  |
+ | FUNCTION_RCDEVICE            | 16384 |
+ | FUNCTION_LIDAR_TF            | 32768 |
+ | FUNCTION_FRSKY_OSD           | 65536 |
+
+Note: `FUNCTION_FRSKY_OSD` = `(1<<16)` requires 17 bits.
+
+### 3. Configuration/MSP Baud Rates
+
+| Baudrate |
+| -------- |
+| 9600     |
+| 19200    |
+| 38400    |
+| 57600    |
+| 115200   |
+| 230400   |
+| 250000   |
+
+### 4. Sensor Input Baud Rates (ESC / GPS)
+
+| Baudrate Identifier | Index |
+| ------------------- | ----- |
+| GPS_BAUDRATE_115200 | 0     |
+| GPS_BAUDRATE_57600  | 1     |
+| GPS_BAUDRATE_38400  | 2     |
+| GPS_BAUDRATE_19200  | 3     |
+| GPS_BAUDRATE_9600   | 4     |
+
+Note: Also has a boolean AUTOBAUD; same values apply for ESC.
+
+### 5. Telemetry Baud Rates
+
+| Baudrate |
+| -------- |
+| AUTO     |
+| 9600     |
+| 19200    |
+| 38400    |
+| 57600    |
+| 115200   |
+
+### 6. Pheripheral Baud Rates
+
+Pheripherals can be Blackbox, VTX, Camera, LIDAR and OSD (FrSky Protocol)
+
+| Baudrate |
+| -------- |
+| AUTO     |
+| 9600     |
+| 19200    |
+| 38400    |
+| 57600    |
+| 115200   |
+| 230400   |
+| 250000   |
+| 1500000  |
+| 2000000  |
+| 2470000  |
+
+### Serial Port Baud Rates
```

Comment:
These are the baud rates for blackbox only - all other peripherals listed in the same drop down as blackbox use fixed baud rates or auto detection only.

Reply:
Maybe baudrate 230400 for openlog should be depreciated in the code because openlog is superseded by openlager anyway?

https://github.com/betaflight/betaflight/blob/master/src/main/blackbox/blackbox_io.c#L297-L305

Reply:
Added missing MSP codes, separated blackbox from other pheripherals, added esc baudrates.